### PR TITLE
Switch to AnalysisContextCollection.

### PR DESCRIPTION
**Pull request by scheglov**

### CODE REVIEW

**Comment by scheglov** [comment]
Hm.. Bots are failing.
But all tests pass locally.

File: lib/src/model/package_builder.dart [hunk 1]
```diff
@@ -200,46 +139,87 @@ class PubPackageBuilder implements PackageBuilder {
   /// Parse a single library at [filePath] using the current analysis driver.
   /// If [filePath] is not a library, returns null.
   Future<DartDocResolvedLibrary> processLibrary(String filePath) async {
-    var name = filePath;
+//    var name = filePath;
```
Comment (srawlins) [inline lib/src/model/package_builder.dart:142]: Delete?

File: lib/src/model/package_builder.dart [hunk 2]
```diff
@@ -200,46 +139,87 @@ class PubPackageBuilder implements PackageBuilder {
...
+    var sourceKind = await session.getSourceKind(filePath);
+
     // Allow dart source files with inappropriate suffixes (#1897).  Those
     // do not show up as SourceKind.LIBRARY.
     if (sourceKind != SourceKind.PART) {
       // Loading libraryElements from part files works, but is painfully slow
       // and creates many duplicates.
-      final library =
-          await driver.currentSession.getResolvedLibrary(source.fullName);
+      print('[processLibrary][filePath: $filePath]');
```
Comment (srawlins) [inline lib/src/model/package_builder.dart:163]: delete debug stuff?

Comment (scheglov) [reply lib/src/model/package_builder.dart:167]: Yeah, of course I will not land it with all this debug output.
But the issue is that it fails on bots, but not locally.
So, I'm trying to debug it :-(
It is tough that I can only change `dartdoc`, while the issue might be somewhere in `analyzer`, and I really want to have debug output in there.

Comment (srawlins) [reply lib/src/model/package_builder.dart]: Can you change dartdoc's dep on analyzer to be a path dependency, and add debug output at will?

Comment (scheglov) [reply lib/src/model/package_builder.dart]: I can change the dependency on analyzer, and I can run tests locally.
They pass locally.
They fail on the bot.
I suspect there are two issues: 
1. with `.dart_tool/build`
2. something turns paths into local case on Windows.
I have had no time to dig deeper again.

**Comment by jcollins-g** [comment]
closing in favor of #2410

**Pull request closed without merging**